### PR TITLE
Update base yamls stubs following IODA refactoring

### LINKED ISSUE #154: Update obsdatain and obsdataout YAML stubs
All `ObsPlugs/*/base/*.yaml` stubs need to be updated following IODA refactoring.  Solution is demonstrated in https://github.com/JCSDA-internal/mpas-jedi/pull/786.

### PR DESCRIPTION
**Pull request by ibanos90**
### Description
This PR updates all `ObsPlugs/*/base/*.yaml` stubs following the refactoring of IODA engines classes and parameters structures according to https://github.com/JCSDA-internal/ioda/pull/776 (adopted solution follows https://github.com/JCSDA-internal/mpas-jedi/pull/786). A new build of the mpas-bundle code is used to account for these modifications. Note that `max frame size` no longer works for obsdataout, therefore the specification for it was removed.

### Issue closed
Closes #154

### Tests completed
 - [x] 3dvar_OIE120km_WarmStart
 - [x] 3denvar_OIE120km_WarmStart
 - [x] 3dvar_OIE120km_ColdStart
 - [x] 3dvar_O30kmIE60km_ColdStart
 - [x] 3denvar_O30kmIE60km_WarmStart
 - [x] eda_OIE120km_WarmStart

### CODE REVIEW

File: config/jedi/ObsPlugs/hofx/base/aircraft.yaml [hunk 1]
```diff
@@ -2,11 +2,14 @@
...
     name: Aircraft
     obsdatain:
-      obsfile: {{InDBDir}}/aircraft_obs_{{thisValidDate}}.h5
+      engine:
+        type: H5File
+        obsfile: {{InDBDir}}/aircraft_obs_{{thisValidDate}}.h5
       max frame size: 40000
     obsdataout:
-      obsfile: {{OutDBDir}}/{{obsPrefix}}_aircraft.h5
-      max frame size: 40000
```
Comment (jjguerrette) [inline config/jedi/ObsPlugs/hofx/base/aircraft.yaml:9]: Do you mind putting a comment in the PR message that `max frame size` no longer works for `obsdataout`?

File: config/jedi/ObsPlugs/hofx/base/iasi_metop-a.yaml [hunk 2]
```diff
@@ -2,10 +2,14 @@
...
+      engine:
+        type: H5File
+        obsfile: {{InDBDir}}/iasi_metop-a_obs_{{thisValidDate}}.h5
       max frame size: 80000
     obsdataout:
-      obsfile: {{OutDBDir}}/{{obsPrefix}}_iasi_metop-a.h5
+      engine:
+        type: H5File
+        obsfile: {{OutDBDir}}/{{obsPrefix}}_iasi_metop-a.h5
       max frame size: 80000
```
Comment (jjguerrette) [inline config/jedi/ObsPlugs/hofx/base/iasi_metop-a.yaml]: Does `max frame size` need to be removed here?

File: config/jedi/ObsPlugs/hofx/base/iasi_metop-b.yaml [hunk 3]
```diff
@@ -2,10 +2,14 @@
...
+      engine:
+        type: H5File
+        obsfile: {{InDBDir}}/iasi_metop-b_obs_{{thisValidDate}}.h5
       max frame size: 80000
     obsdataout:
-      obsfile: {{OutDBDir}}/{{obsPrefix}}_iasi_metop-b.h5
+      engine:
+        type: H5File
+        obsfile: {{OutDBDir}}/{{obsPrefix}}_iasi_metop-b.h5
       max frame size: 80000
```
Comment (jjguerrette) [inline config/jedi/ObsPlugs/hofx/base/iasi_metop-b.yaml]: Does `max frame size` need to be removed here?

File: config/jedi/ObsPlugs/hofx/base/iasi_metop-c.yaml [hunk 4]
```diff
@@ -2,10 +2,14 @@
...
+      engine:
+        type: H5File
+        obsfile: {{InDBDir}}/iasi_metop-c_obs_{{thisValidDate}}.h5
       max frame size: 80000
     obsdataout:
-      obsfile: {{OutDBDir}}/{{obsPrefix}}_iasi_metop-c.h5
+      engine:
+        type: H5File
+        obsfile: {{OutDBDir}}/{{obsPrefix}}_iasi_metop-c.h5
       max frame size: 80000
```
Comment (jjguerrette) [inline config/jedi/ObsPlugs/hofx/base/iasi_metop-c.yaml]: Does `max frame size` need to be removed here?

File: config/jedi/ObsPlugs/variational/base/iasi_metop-a.yaml [hunk 5]
```diff
@@ -2,10 +2,14 @@
...
+      engine:
+        type: H5File
+        obsfile: {{InDBDir}}/iasi_metop-a_obs_{{thisValidDate}}.h5
       max frame size: 80000
     obsdataout:
-      obsfile: {{OutDBDir}}{{MemberDir}}/{{obsPrefix}}_iasi_metop-a.h5
+      engine:
+        type: H5File
+        obsfile: {{OutDBDir}}{{MemberDir}}/{{obsPrefix}}_iasi_metop-a.h5
       max frame size: 80000
```
Comment (jjguerrette) [inline config/jedi/ObsPlugs/variational/base/iasi_metop-a.yaml]: Does `max frame size` need to be removed here?

File: config/jedi/ObsPlugs/variational/base/iasi_metop-b.yaml [hunk 6]
```diff
@@ -2,10 +2,14 @@
...
+      engine:
+        type: H5File
+        obsfile: {{InDBDir}}/iasi_metop-b_obs_{{thisValidDate}}.h5
       max frame size: 80000
     obsdataout:
-      obsfile: {{OutDBDir}}{{MemberDir}}/{{obsPrefix}}_iasi_metop-b.h5
+      engine:
+        type: H5File
+        obsfile: {{OutDBDir}}{{MemberDir}}/{{obsPrefix}}_iasi_metop-b.h5
       max frame size: 80000
```
Comment (jjguerrette) [inline config/jedi/ObsPlugs/variational/base/iasi_metop-b.yaml]: Does `max frame size` need to be removed here?

File: config/jedi/ObsPlugs/variational/base/iasi_metop-c.yaml [hunk 7]
```diff
@@ -2,10 +2,14 @@
...
+      engine:
+        type: H5File
+        obsfile: {{InDBDir}}/iasi_metop-c_obs_{{thisValidDate}}.h5
       max frame size: 80000
     obsdataout:
-      obsfile: {{OutDBDir}}{{MemberDir}}/{{obsPrefix}}_iasi_metop-c.h5
+      engine:
+        type: H5File
+        obsfile: {{OutDBDir}}{{MemberDir}}/{{obsPrefix}}_iasi_metop-c.h5
       max frame size: 80000
```
Comment (jjguerrette) [inline config/jedi/ObsPlugs/variational/base/iasi_metop-c.yaml]: Does `max frame size` need to be removed here?

**Review by jjguerrette — CHANGES_REQUESTED** [review]
Only a few minor changes/comments.  Waiting for tests to be updated before approving.  Thank you for the quick turnaround @ibanos90.

**Comment by ibanos90** [comment]
> Only a few minor changes/comments. Waiting for tests to be updated before approving. Thank you for the quick turnaround @ibanos90.

Thanks for your review @jjguerrette! I will remove the `max frame size` from those yamls.

File: config/jedi/ObsPlugs/hofx/base/aircraft.yaml (hunk 1 above)
Comment (ibanos90) [reply config/jedi/ObsPlugs/hofx/base/aircraft.yaml:9]: That's a good point. It would be good for future reference. Thanks.

**Comment by jjguerrette** [comment]
Did the tests finish @ibanos?  I'm going to merge your branch into my departure sign change branch in order to run my own tests.

**Comment by ibanos90** [comment]
> Did the tests finish @ibanos? I'm going to merge your branch into my departure sign change branch in order to run my own tests.

Hi @jjguerrette! Most of the tests did, but I am getting an error in the tests with ColdStart:
```
Bad parameter: /: Mandatory parameter 'sensor' not found caught in  (/glade/p/mmm/parc/ivette/pandac/jedi/22SEP2022/mpas-bundle/oops/src/oops/runs/Run.cc +192 execute)
Exception: oops::Variational<MPAS, UFO and IODA observations> terminating...
Exception stack: 
Bad parameter: /: Mandatory parameter 'sensor' not found
backtrace [5] stack has 24 addresses
```
I am double checking the yaml files.

**Comment by ibanos90** [comment]
Hi @jjguerrette, all the tests now complete successfully after adding the `sensor` specification for the `ObsErrorFactorSurfJacobianRad` observations function.

**Comment by jjguerrette** [comment]
> all the tests now complete successfully after adding the `sensor` specification for the `ObsErrorFactorSurfJacobianRad` observations function.

Just a formality, but please check the boxes before merging.

**Review by jjguerrette — APPROVED** [review]

**Pull request merged**